wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/environment/server/compose/deserializer/EnvironmentDeserializer.java

```diff
@@ -85,4 +83,21 @@
         */
         throw ctxt.mappingException(format("Unsupported type '%s'.", environment.getClass()));
     }
+
+    /**
+     * Parse "key=value" item.
+     * @param item - environment array item to parse
+     * @return map of key/value which are divided by "=" sign in item.
+     * @throws IllegalArgumentException if there is no "=" sign in item, or if there is empty key in item
+     */
+    private Map<String, String> parseEnvironmentItem(String item) throws IllegalArgumentException {
+        if (!item.contains(ARRAY_ITEM_DIVIDER) || item.indexOf(ARRAY_ITEM_DIVIDER) == 0) {
+            throw new IllegalArgumentException(format(UNSUPPORTED_VALUE_MESSAGE, item));
+        }
+
+        String key = item.substring(0, item.indexOf(ARRAY_ITEM_DIVIDER));
```

Comment:
I think it is better to use single `indexOf` call instead of `2 indexOf + contains`